packages/frontend/app/root.tsx

```diff
@@ -184,6 +208,15 @@ export function ErrorBoundary() {
   let errorMessage = 'Unknown error'
   if (error instanceof Error) {
     errorMessage = error.message
+
+    if (
+      isApolloError(error) &&
+      error.networkError &&
+      'statusCode' in error.networkError &&
+      error.networkError.statusCode === 401
+    ) {
+      errorMessage = 'Unauthorized. Please set your API credentials.'
+    }
```

Comment:
The behavior on access a route with missing or bad api credentials is handled here. Clicking the navigation is disabled but if you manually go to /assets for example you'll see our regular error screen with this unauthorized message and prompt to visit the homepage, where the api credentials form lives.

![image](https://github.com/user-attachments/assets/7365074b-8146-44da-9fb1-dcb88e7a9859)

Reply:
Parsing this is kinda of a PITA but I think this is the correct way. It's an apollo error but we are actually throwing the error in koa so it ends up as a network error (which could be one of many types, hence `'statusCode' in error.networkError`).

AFAIK this should be reliable but the fact that `networkError` could be one of many types does make you wonder. I also considered simply doing 
 `if(error.message.includes('401'))`. This works but I figured that's likely flakier.

Reply:
actually this isnt right.... not correctly detecting it and showing the message I intended. reverting back to checking the error message.